app/src/main/java/com/islamey/hamza/wazaker/ui/CounterFragment/CounterFragment.kt

```diff
@@ -1,86 +1,68 @@
 package com.islamey.hamza.wazaker.ui.CounterFragment
 
 import android.annotation.SuppressLint
-import android.content.Context
-import android.content.SharedPreferences
-import android.graphics.Color
 import android.os.Bundle
 import android.view.*
 import android.widget.Toast
 import androidx.fragment.app.Fragment
-import com.islamey.hamza.wazaker.utils.RandomColor
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.islamey.hamza.wazaker.utils.RandomColor.getRandomColor
 import com.islamey.wazkar.R
 import com.islamey.wazkar.databinding.FragmentCounterBinding
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
 
+@AndroidEntryPoint
 class CounterFragment : Fragment() {
 
-    private var _binding: FragmentCounterBinding? = null
-    private val binding get() = _binding!!
+    private lateinit var binding: FragmentCounterBinding
 
-    private val MY_PREFS_NAME = "pref"
-    private var sharedPreferences: SharedPreferences? = null
-
-    private var counts = 0
-    private var totalCounts = 0
+    private val counterViewModel: CounterViewModel by viewModels()
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
+        setHasOptionsMenu(true)
+        binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
 
-        _binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
-
-        getDataFromSharedPreferences()
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-        binding.counterscreen.setOnClickListener {
-            counts++
-            totalCounts++
+        initUi()
+    }
 
-            checkCounts(counts)
-            binding.countertextview.text = counts.toString()
-            setTotalZekerCountstext(totalCounts)
+    @SuppressLint("SetTextI18n")
```

Comment:
do we need this suppress ? 
if yes, we can also, use data binding and get value into xml from view model as we will do for listener